src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/configuration/KendraIntelligentRankingConfiguration.java

```diff
@@ -76,7 +77,7 @@ public void writeTo(StreamOutput out) throws IOException {
     this.properties.writeTo(out);
   }
 
-  public static ResultTransformerConfiguration parse(XContentParser parser) throws IOException {
+  public static KendraIntelligentRankingConfiguration parse(XContentParser parser) throws IOException {
```

Comment:
re:naming - is this going from less specific naming to more specific intentionally?

Reply:
Yes.

a) It helps the unit tests, since we can confirm the specific type.
b) There's a best-practice to take the most general type of argument and return the most specific type. I think it's an application of the [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).